microsoft-edge/devtools-guide-chromium/memory-problems/dom-leaks.md

```diff
@@ -40,92 +41,128 @@ As explained in [Fix memory problems](index.md), memory issues affect page perfo
 
-1. In the **Memory** tool, select the link **room.js:13**.
+   The **Memory** tool automatically selects the object in the heap that is referencing the detached element.  Such an object is called a _retainer_.
+
+1. In the **Retainers** tab, in a retainer sub-entry about the `unmounted` member, such as **unmounted in Room @54011**, click the link **room.js:13**.
+
+   The **Sources** tool opens in the **Activity Bar** and shows line 13 of the file **room.js** (the `Room` constructor):
 
-   The **Sources** tool opens in the **Activity Bar** and shows line 13 of the file **room.js**.
+   ![Line 13 in room.js: the Room constructor](./dom-leaks-images/room-js-line-13.png)
 
-1. In the `hide()` function of **room.js**, the JavaScript code of the demo application adds each message in the room to an `unmounted` array. The `unmounted` array is the object that is referencing the detached element:
+   `unmounted` is an array member of the `Room` class, defined in line 19 in the constructor: `this.unmounted = [];`.
+
+1. Scroll down to line 49, `this.unmounted.push(el);` within the `hide` method of the `Room` class:
 
    ![Identifying the JavaScript that is retaining the detached element](./dom-leaks-images/javascript-retainers.png)
 
-You have now identified the retainer that is preventing the detached element from being garbage-collected by the browser!
+   The code adds each message in the room to the `unmounted` array.  The `unmounted` array is the object that is referencing the detached element.
+
+In the JavaScript code, you have now identified the retainer object (the `unmounted` array) that's preventing the detached element from being garbage-collected by the browser.
+
+You've found detached elements that the browser can't garbage-collect, and located the JavaScript object that's still referencing the detached element.  You can then change the JavaScript code to release the element, to reduce the number of detached elements on the webpage, increasing webpage performance and responsiveness.
+
+
+<!-- copied h4 section to article "Debug DOM memory leaks ("Detached elements" profiling type)" -->
+<!-- ------------------------------ -->
+#### Fixing the JavaScript code to prevent detached elements that can't be garbage-collected
+
+The `collectOldMessages` method of the `Room` class (on line 26) contains comments about a potential leak, and recommends making a single cache for all `Room` instances, instead of a cache for each `Room` instance, and then monitoring how many elements get into that cache over time.
 
+There is a potential leak in the existing code, because the cleanup in this method occurs at a different rate than the addition of new messages.  We can easily find ourselves in a situation where we have messages in the cache that don't need (or can't) be reused right away.  This is not likely to become too much, but the cache is per-room, which multiplies the problem.
```

Comment:
I removed entire h4 section; you wrote remove "section" not remove "paragraph", @captainbrosset 
DE tool article > h4: Fixing the JavaScript code to prevent detached elements that can't be garbage-collected

contained 2 paras:
```
The `collectOldMessages` method of the `Room` class (on line 26) contains comments about a potential leak, and recommends making a single cache for all `Room` instances, instead of a cache for each `Room` instance, and then monitoring how many elements get into that cache over time.

There is a potential leak in the existing code, because the cleanup in this method occurs at a different rate than the addition of new messages.  We can easily find ourselves in a situation where we have messages in the cache that don't need (or can't) be reused right away.  This is not likely to become too much, but the cache is per-room, which multiplies the problem.
```